docs/core/design/dcrypt.md

```diff
@@ -109,30 +115,206 @@ asymmetric key pair. File encryption can be done using whatever
+
+To simulate this scenario consider storing an input file `$in_file` using:
+- `$private_key_path` and `$public_key_path` - the private and public key files
+  to allow encryption with,
+- `$prefix` - the path prefix to use for storing the output file,
+- `$out_file` - the file path to store the file as.
+
+```sh
+doveadm fs put compress \
+    gz:6:crypt:private_key_path=$private_key_path:public_key_path=$public_key_path:posix:prefix=$prefix \
+    $in_file $out_file
+```
+
+Using the `dcrypt-decrypt` script you can then retrieve the key and file
+information using the following command. Make sure to supply the same
+`$private_key_path`, `$prefix` and `$out_file` as above:
+
+```sh
+dcrypt-decrypt.py -i -k $private_key_path -f ${prefix}${out_file} | gunzip
+```
+
+::: info
+The piped `gunzip` will ensure the contents are decompressed after decryption.
+:::
+
+### Output Description
+
+Run `dcrypt-decrypt.py --help` to read the tool's help text, which gives hints
+about its supported flags and its output.
+
+Supplying the private key file via the `-k`/`--key` flag will decrypt the
+encrypted file contents.
+
+::: info
+Fields that contain binary data are represented in hexadecimal form. Other
+fields are converted or displayed in human readable form.
+:::
+
+#### File information
+
+The `-i`/`--info` flag outputs a list of fields of the supplied file:
+
+`Version`
+:   What version of the dovecot [[link,lib_dcrypt_key_formats,Key Formats]] has
+    been used to encrypt this file.
+
+`Flags`
+:   What [[link,lib_dcrypt_flags,Flags]] have been used when encrypting the
+    file.
+
+`Header length`
+:   Number of bytes in the header.
+
+`Cipher algo`
+:   Name and Object Identifier (OID) of the cipher used for encrypting the
+    key - and data if it was used for both.
+
+`Digest algo`
+:   Name and OID of the digest algorithm used for the key and data.
+
+Key(s) used for encrypting the file
+
+`Key type`
+:   Either `EC` for Elliptic-Curve or `RSA` for Rivest-Shamir-Adleman key
+    type.
+
+`Key digest`
+:   Hash of public key id in Distinguished Encoding Rules (DER) format.
+
+`Peer key`
+:   Ephemeral key used to derive shared secret used for key material
+    encryption.
+
+`Encrypted`
+:   The encrypted encryption key material.
+
+`Kd hash`
+:   The checksum of the key data.
+
+#### Key information
+
+The following information are only available if the `-k`/`--key` flag was
+supplied with the matching key.
+
+`Provided key`
+:   Checksum of the key that was provided.
+
+##### Key derivation data
+
+`Key derivation`
+:   General information regarding the supplied key file.
+
+With its attributes:
+
+`Rounds`
+:   Number of iterations that the digestion algorithm is repeated.
+
+`Secret`
+:   Decrypted secret from the peer key.
+
+`Salt`
+:   Salt used for hashing. `Peer key` is used here.
+
+##### Encryption and Decryption data
+
+`Encryption key decryption`
+:   General information regarding the key used to encrypt the symmetric key
```

Comment:
The encryption is always symmetric, the key derivation just changes. For the KEM part we use the shared secret and peer key to derive the symmetric key and for data we use the decapsulated random data for decrypting the payload.

Reply:
I used the words you used here and slightly changed this and the `Key` attribute further below.
Sorry that this is so confused, but I barely understand what is being done, let alone what the words mean. :astonished: